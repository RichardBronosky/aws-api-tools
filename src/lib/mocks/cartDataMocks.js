export const mockCartItem1 = {
  activeAddons: null,
  price: 279.99,
  product: {
    title: 'Adaire Brown Side Table',
    sku: '23122360',
    price: 279.99,
    list_price: 349.99,
    delivery_type: 'O',
    category: 'tables',
    free_shipping: true,
    on_sale: true,
    primary_image:
      'http://images.rtg-dev.com/adaire-brown-side-table_23122360_image-item?cache-id=81c2298dd5804701ca610022d0a4d090',
    shipping_cost_code: { FL: 'VD71', SE: 'VD71', TX: 'VD71' },
    strikethrough: { FL: true, OM: true, SE: true, TX: true },
  },
  sku: '23122360',
  quantity: 1,
}

export const mockCartItem2 = {
  sku: '1141677P',
  product: {
    title: 'Laney Park Dark Gray 7 Pc Sectional',
    sku: '1141677P',
    price: 1199.99,
    list_price: 1299.99,
    delivery_type: 'D',
    category: 'livingroom',
    free_shipping: false,
    on_sale: true,
    primary_image:
      'http://images.rtg-dev.com/laney-park-dark-gray-7-pc-sectional_1141677P_image-room?cache-id=a7e194ec3ebbb82c6482c801d3971efc',
    package_skus: {
      SE: [
        { sku: '12216760', quantity: 1, price: '208.43' },
        { sku: '12116768', quantity: 4, price: '172.94' },
        { sku: '12316762', quantity: 1, price: '231.63' },
        { sku: '12516766', quantity: 1, price: '68.17' },
      ],
      TX: [
        { sku: '12316762', quantity: '1', price: '231.63' },
        { sku: '12216760', quantity: '1', price: '208.43' },
        { sku: '12516766', quantity: '1', price: '68.17' },
        { sku: '12116768', quantity: '4', price: '172.94' },
      ],
      FL: [
        { sku: '12316762', quantity: '1', price: '248.65' },
        { sku: '12116768', quantity: '4', price: '185.62' },
        { sku: '12216760', quantity: '1', price: '223.71' },
        { sku: '12516766', quantity: '1', price: '73.16' },
      ],
    },
    shipping_cost_code: { FL: 'ROOM', SE: 'ROOM', TX: 'ROOM' },
  },
  price: 1199.99,
  activeAddons: null,
  quantity: 3,
}

export const mockCart = {
  cartItems: [mockCartItem1, mockCartItem2],
}
