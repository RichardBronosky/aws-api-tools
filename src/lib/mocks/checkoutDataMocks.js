export const order1 = {
  isPickup: false,
  isExpress: false,
  submitted: false,
  emailSubscribed: true,
  additionalDirections: 'test',
  lineItems: [
    {
      sku: '26043632',
      quantity: 2,
      unitPrice: 174,
      title: 'Collins Point Brown Accent Chair',
      deliveryType: 'O',
      category: 'livingroom',
      childItems: [],
    },
    {
      sku: '26043633',
      quantity: 2,
      unitPrice: 174,
      title: 'Collins Point Brown Accent Chair2',
      deliveryType: 'D',
      category: 'livingroom',
      childItems: [],
    },
    {
      sku: '26043634',
      quantity: 2,
      unitPrice: 174,
      title: 'Collins Point Brown Accent Chair3',
      deliveryType: 'U',
      category: 'livingroom',
      childItems: [],
    },
    {
      category: 'gift-card',
      childItems: [],
      deliveryType: 'T',
      quantity: 1,
      sku: '83333333',
      title: 'Gift Card',
      unitPrice: 100,
    },
  ],
  createdAt: '2019-05-29T15:20:15.170Z',
  lastModified: '2019-05-29T17:00:09.305Z',
  calendarType: 'daily',
  dDeliveryCharge: 0,
  deliveryCalendar: [],
  kDeliveryCharge: 0,
  paymentInfo: [
    {
      authorized: false,
      paymentType: 'CYBERV2',
      authorizedAmount: 377.58,
      paymentProperties: {
        postback: {
          req_card_type: '001',
          decision: 'ACCEPT',
        },
      },
    },
  ],
  pickupCalendar: [],
  pickupWarehouse: '',
  promotions: {
    cartTotal: 348,
    listPriceTotal: 348,
    salePriceTotal: 348,
    bonusBuyTotal: 0,
    totalSavings: 0,
    lineItems: [
      {
        sku: '26043632',
        quantity: 2,
        unitPrice: 174,
        title: 'Collins Point Brown Accent Chair',
        deliveryType: 'O',
        category: 'livingroom',
        childItems: [],
      },
    ],
    promotions: [],
  },
  total: 377.58,
  totalDeliveryCharge: 0,
  contact: {
    firstName: 'Tester',
    lastName: 'Test',
    email: 'test@test.com',
    altPhone: '',
    phone: '4040404040',
  },
  payer: {
    firstName: 'Tester',
    lastName: 'Test',
    email: 'test@test.com',
    altPhone: '',
    phone: '4040404040',
    billingDifferent: true,
    billingSubmitted: false,
  },
  deliveryTexts: true,
  distributionIndex: 1,
  emailCampaign: true,
  shippingAddress: {
    address1: '103 seffner ave',
    address2: '',
    city: 'seffner',
    state: 'fl',
    zip: '33584',
    county: 'Hillsborough',
    country_code: 'US',
    addressLookup: '',
    addressLookupSuccess: false,
    showAddressLookup: false,
  },
  billingAddress: {
    address1: '103 seffner ave',
    address2: '',
    city: 'seffner',
    state: 'fl',
    zip: '33584',
    county: 'Hillsborough',
    country_code: 'US',
    addressLookup: '',
    addressLookupSuccess: false,
    showAddressLookup: false,
  },
  additionalDirections: '',
  deliveryDate: '2019-06-04',
  shouldCombineUPSWithTruck: true,
  tax: 29.58,
  amountDue: 0,
  cyberSourceSignature: {
    bill_to_forename: 'Tester',
    bill_to_surname: 'Test',
    bill_to_email: 'test@test.com',
    bill_to_address_line1: '103 seffner ave',
    bill_to_address_city: 'seffner',
    bill_to_address_postal_code: '33584',
    bill_to_address_state: 'FL',
    bill_to_address_country: 'US',
    amount: '377.58',
    currency: 'USD',
    reference_number: '9d06c1c1-ea10-4b07-82e0-27ae4947e6ef',
    locale: 'en-us',
    signed_date_time: '2019-05-29T16:59:55Z',
    profile_id: 'C43A2464-4A37-4A37-BF2C-6E62CFCEBA41',
    access_key: '66674d4c4897393fb6941963f6ac6d1a',
    transaction_uuid: 'd9aa39ff-793c-4157-b017-853ed824ad82',
    transaction_type: 'create_payment_token',
    unsigned_field_names: 'payment_method',
    payment_method: 'card',
    item_0_name: '26043632',
    item_0_sku: '26043632',
    item_0_quantity: 2,
    item_0_unit_price: '377.58',
    item_0_taxAmount: '0.00',
    line_item_count: 1,
    signed_field_names:
      'bill_to_forename,bill_to_surname,bill_to_email,bill_to_address_line1,bill_to_address_city,bill_to_address_postal_code,bill_to_address_state,bill_to_address_country,amount,currency,reference_number,locale,signed_date_time,profile_id,access_key,transaction_uuid,transaction_type,unsigned_field_names,item_0_name,item_0_sku,item_0_quantity,item_0_unit_price,item_0_taxAmount,line_item_count,signed_field_names',
    signature: 'T0y65CK+W73H2uk483GgOHGVWSQmsPSZs3Hi1RR0NKo=',
  },
  giftCardInfo: {
    giftCards: [],
    useGiftCard: false,
    giftCardNumber: '',
    giftCardPin: '',
    addAnotherCard: false,
  },
  orderId: '9d06c1c1-ea10-4b07-82e0-27ae4947e6ef',
}

export const mockDeliveryCalendar = [
  {
    date: '2019-06-04',
    readable: 'June 04',
    dayOfWeek: 'Tue',
    isPickup: true,
    isStandardDelivery: true,
    isExpressDelivery: false,
  },
  {
    date: '2019-06-05',
    readable: 'June 05',
    dayOfWeek: 'Wed',
    isPickup: true,
    isStandardDelivery: true,
    isExpressDelivery: false,
  },
  {
    date: '2019-06-06',
    readable: 'June 06',
    dayOfWeek: 'Thu',
    isPickup: true,
    isStandardDelivery: true,
    isExpressDelivery: false,
  },
  {
    date: '2019-06-07',
    readable: 'June 07',
    dayOfWeek: 'Fri',
    isPickup: true,
    isStandardDelivery: true,
    isExpressDelivery: false,
  },
]

export const fetchAddressLookupResp = {
  totalMatches: 136009,
  count: 9,
  results: [
    {
      suggestion: '4100 Alexandria Pike, Alexandria KY 41001',
      matched: [[36, 40], [0, 4]],
      format:
        'https://api.edq.com/capture/address/v2/format?country=USA&id=15197f6d-d4b7-41ad-bc77-b0d5790baa0a_U40_24_0_0_0%3D4100%24QL4%24GEO0',
    },
    {
      suggestion: '4100 Amy Lou Dr, Alexandria KY 41001',
      matched: [[31, 35], [0, 4]],
      format:
        'https://api.edq.com/capture/address/v2/format?country=USA&id=420547b1-8fac-4d4d-bc9c-776fc6a7b6ff_U40_24_0_0_0%3D4100%24QL4%24GEO0',
    },
    {
      suggestion: '4100 Beiting Dr, Alexandria KY 41001',
      matched: [[31, 35], [0, 4]],
      format:
        'https://api.edq.com/capture/address/v2/format?country=USA&id=61f7eaae-0ad7-4f7f-8cbe-68e126ad190e_U40_24_0_0_0%3D4100%24QL4%24GEO0',
    },
    {
      suggestion: '4100 Ky 915, Alexandria KY 41001',
      matched: [[27, 31], [0, 4]],
      format:
        'https://api.edq.com/capture/address/v2/format?country=USA&id=204b8965-23a8-4d68-a296-5c17e03cd105_U40_24_0_0_0%3D4100%24QL4%24GEO0',
    },
    {
      suggestion: '4100 Licking Pike, Alexandria KY 41001',
      matched: [[33, 37], [0, 4]],
      format:
        'https://api.edq.com/capture/address/v2/format?country=USA&id=ed19c1cb-ce4d-47d8-b967-50f3f43cd3b0_U40_24_0_0_0%3D4100%24QL4%24GEO0',
    },
    {
      suggestion: '4100 Old State Road 2, Alexandria KY 41001',
      matched: [[37, 41], [0, 4]],
      format:
        'https://api.edq.com/capture/address/v2/format?country=USA&id=f67f813f-97e5-4683-9274-579267c53238_U40_24_0_0_0%3D4100%24QL4%24GEO0',
    },
    {
      suggestion: '4100 Persimmon Grove Rd, Alexandria KY 41001',
      matched: [[39, 43], [0, 4]],
      format:
        'https://api.edq.com/capture/address/v2/format?country=USA&id=5cb5e9d7-8cd5-49e0-b2eb-be92c95ca940_U40_24_0_0_0%3D4100%24QL4%24GEO0',
    },
    {
      suggestion: '4100 Sr 10, Alexandria KY 41001',
      matched: [[26, 30], [0, 4]],
      format:
        'https://api.edq.com/capture/address/v2/format?country=USA&id=4edb7aa7-527d-4c0d-a4c7-5534b858e7ac_U40_24_0_0_0%3D4100%24QL4%24GEO0',
    },
    {
      suggestion: '4100 State Route 10, Alexandria KY 41001',
      matched: [[35, 39], [0, 4]],
      format:
        'https://api.edq.com/capture/address/v2/format?country=USA&id=90b44538-006a-488a-bf80-6e0517808017_U40_24_0_0_0%3D4100%24QL4%24GEO0',
    },
  ],
}

export const giftCards1 = [
  {
    giftCardNumber: '1111111111111111',
    giftCardPin: '111111',
    authorizedAmount: 100,
    balance: 200,
  },
]
