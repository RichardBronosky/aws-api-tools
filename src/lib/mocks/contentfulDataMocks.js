export const Banner1 = {
  contentfulContentGroup: {
    groupContent: [
      {
        __typename: 'ContentfulBanner',
        title: 'Promo E : Umbrella Sale 1 - C6 R1',
        startDate: null,
        endDate: null,
        displayContent: true,
        contentText: {
          childMarkdownRemark: {
            html:
              '<br>\n<span style="color:#FFFFFF; font-weight: bold; font-size: 28px;">GET A FREE UMBRELLA</span>\n<br>\n<span style="color:#FFFFFF; font-weight: bold; font-size: 28px;">WITH SELECT* OUTDOOR PURCHASES OF $999+</span>\n<br>\n<br>\n<span style="color:#FFFFFF; font-weight: bold;">*EXCLUDES VENDOR DIRECT.</span>\n<br>',
          },
        },
        contentHorizontalRuleColor: null,
        textPosition: 'center',
        contentPosition: 'center',
        containerSize: '50%',
        contentLinkColor: 'white',
        link: {
          id: '12703942-b7d5-5bc8-b79b-ad18d5321106',
          text: 'Link : Outdoor Patio',
          displayText: 'Outdoor',
          textColor: null,
          internalUrl: null,
          slug: '/furniture/outdoor-patio',
          url: null,
          altDesc: 'View All Outdoor Patio Furniture',
          title: null,
          type: null,
          target: null,
        },
        contentLink: null,
        buttons: null,
        imageAltText: 'Shop Umbrella Sale',
        desktopBackgroundImage: null,
        backgroundColor: 'Blue',
        mobileImage: {
          id: '4e8177a3-6c27-5544-a5ee-fcb384528020',
          sizes: {
            srcSet:
              '//images.ctfassets.net/n86sd31cg9pg/sJQJZln6DvDFCsfwnBojB/53c4bc3267589bf8fcba15f3a23417e3/AnnSale_Umbrella1_MOBILE_640x600.png?w=640&h=600&q=50 640w',
            src:
              '//images.ctfassets.net/n86sd31cg9pg/sJQJZln6DvDFCsfwnBojB/53c4bc3267589bf8fcba15f3a23417e3/AnnSale_Umbrella1_MOBILE_640x600.png?q=50',
            sizes: '(max-width: undefinedpx) 100vw, undefinedpx',
          },
          file: {
            url:
              '//images.ctfassets.net/n86sd31cg9pg/sJQJZln6DvDFCsfwnBojB/53c4bc3267589bf8fcba15f3a23417e3/AnnSale_Umbrella1_MOBILE_640x600.png',
            fileName: 'AnnSale_Umbrella1_MOBILE_640x600.png',
            contentType: 'image/png',
          },
          resolutions: { width: null, height: null },
        },
        desktopImage: {
          id: '8740a3af-1f18-5f98-9335-137b1f18a7c7',
          sizes: {
            srcSet:
              '//images.ctfassets.net/n86sd31cg9pg/7t1ifBRd9ZaAb7NssXH9KT/3ce7539777e89b8446c591144e989c76/AnnSale_Umbrella1_CONTAINER_1660x350.png?w=1660&h=350&q=50 1660w',
            src:
              '//images.ctfassets.net/n86sd31cg9pg/7t1ifBRd9ZaAb7NssXH9KT/3ce7539777e89b8446c591144e989c76/AnnSale_Umbrella1_CONTAINER_1660x350.png?q=50',
            sizes: '(max-width: undefinedpx) 100vw, undefinedpx',
          },
          file: {
            url:
              '//images.ctfassets.net/n86sd31cg9pg/7t1ifBRd9ZaAb7NssXH9KT/3ce7539777e89b8446c591144e989c76/AnnSale_Umbrella1_CONTAINER_1660x350.png',
            fileName: 'AnnSale_Umbrella1_CONTAINER_1660x350.png',
            contentType: 'image/png',
          },
          resolutions: { width: null, height: null },
        },
        contentPadding: null,
        contentful_id: '3fezOgXNiGTW87ngJIx9cB',
      },
    ],
  },
}
