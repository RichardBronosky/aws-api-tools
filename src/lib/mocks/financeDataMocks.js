export const mockFinancePlans1 = [
  {
    financeCode: '50EQUALPMTS',
    effectiveDate: '2019-06-03 00:00:00',
    creditEndsDate: '2023-08-01 00:00:00',
    downPaymentRequired: true,
    termsAndConditions: {
      childMarkdownRemark: {
        html:
          'Finance Plan:<br><br>Promotion Type: <strong>No Interest until August 2023*</strong><br>Promotion Duration: <strong>Until August 2023*</strong><br>Promotional APR: 0%<br>Purchase APR: 29.99% <br>Down payment of sales tax and delivery required<br><br><br><br><strong>No Interest/Equal Payments</strong><br>On promotional purchase, monthly payments equal to the promotional purchase divided equally by the number of months in the promotional period are required until expiration or termination of promotion, but no Finance Charges will be assessed during Promotion Duration indicated if all minimum monthly payments on account paid when due.<br>The Purchase APR as indicated applies to expired promotions and optional charges. If minimum monthly payments are 60 days past due, the promotion may be terminated and a Penalty APR may apply. Promotional purchases on existing accounts may not receive full benefit of promotion terms, including reduced APR if applicable, if account is subject to penalty pricing. Payments over the minimum will be applied as required by applicable law. If you receive a change in terms of penalty rate notice that becomes effective before the delivery/installation date of your purchase, the new Purchase APR or Penalty APR (as applicable) will apply to the purchase balance added to your account upon delivery/installation. Subject to approval by Synchrony Bank.<br>',
      },
    },
    priority: '1',
    promoMessage: {
      childMarkdownRemark: {
        html:
          '<p><b>No Interest until August 2023*</b></p><p>On purchases priced at $499.99 and up made with your Rooms To Go credit card through 6/17/19. Equal Monthly Payments required until August 2023*.</p><p>Rooms To Go requires a down payment equal to sales tax and delivery. </p><p>*Promotional Offer applies only to single-receipt qualifying purchases of $499.99 and up. No interest until June 2021 on single-receipt qualifying purchases less than $499.99. No interest will be charged on promo purchase (including related optional credit insurance/debt cancellation charges) and equal monthly payments are required equal to initial promo purchase amount divided equally by the number of months in promo period until promo is paid in full. The equal monthly payment will be rounded to the next highest whole dollar and may be higher than the minimum payment that would be required if the purchase was a non-promotional purchase. Any monthly payments shown are in connection with the 60 months promotional offer, exclude taxes and delivery, and should allow you to pay off the promotional purchase within the promo period if (1) you make your payments by the due date each month and (2) this is the only balance on your account during the promo period.  If you have other balances on your account, this monthly payment will be added to the minimum payment applicable to those balances.  Regular account terms apply to non-promotional purchases. For new accounts: Purchase APR is 29.99. Minimum Interest Charge is $2. Existing cardholders should see their credit card agreement for their applicable terms. Subject to credit approval.</p><p>Rooms To Go requires a down payment equal to sales tax and delivery. </p>',
      },
    },
    threshold: '499.99',
    numberOfMonths: '50',
    siteFriendlyLabel: 'No Interest until August 2023*',
  },
  {
    financeCode: '90015-0621',
    effectiveDate: '2019-06-03 00:00:00',
    creditEndsDate: '2021-06-01 00:00:00',
    downPaymentRequired: true,
    termsAndConditions: {
      childMarkdownRemark: {
        html:
          'Finance Plan:<br><br>Promotion Type: <strong>No Interest until June 2021*</strong><br>Promotion Duration: <strong>Until June 2021*</strong><br>Promotional APR: 0%<br>Purchase APR: 29.99% <br>Down payment of sales tax and delivery required<br><br><br><br><strong>No Interest/Equal Payments</strong><br>On promotional purchase, monthly payments equal to the promotional purchase divided equally by the number of months in the promotional period are required until expiration or termination of promotion, but no Finance Charges will be assessed during Promotion Duration indicated if all minimum monthly payments on account paid when due.<br>The Purchase APR as indicated applies to expired promotions and optional charges. If minimum monthly payments are 60 days past due, the promotion may be terminated and a Penalty APR may apply. Promotional purchases on existing accounts may not receive full benefit of promotion terms, including reduced APR if applicable, if account is subject to penalty pricing. Payments over the minimum will be applied as required by applicable law. If you receive a change in terms of penalty rate notice that becomes effective before the delivery/installation date of your purchase, the new Purchase APR or Penalty APR (as applicable) will apply to the purchase balance added to your account upon delivery/installation. Subject to approval by Synchrony Bank.<br>',
      },
    },
    priority: '0',
    promoMessage: {
      childMarkdownRemark: {
        html:
          '<p><b>No Interest until August 2023*</b></p><p>On purchases priced at $499.99 and up made with your Rooms To Go credit card through 6/17/19. Equal Monthly Payments required until August 2023*.</p><p>Rooms To Go requires a down payment equal to sales tax and delivery. </p><p>*Promotional Offer applies only to single-receipt qualifying purchases of $499.99 and up. No interest until June 2021 on single-receipt qualifying purchases less than $499.99. No interest will be charged on promo purchase (including related optional credit insurance/debt cancellation charges) and equal monthly payments are required equal to initial promo purchase amount divided equally by the number of months in promo period until promo is paid in full. The equal monthly payment will be rounded to the next highest whole dollar and may be higher than the minimum payment that would be required if the purchase was a non-promotional purchase. Any monthly payments shown are in connection with the 60 months promotional offer, exclude taxes and delivery, and should allow you to pay off the promotional purchase within the promo period if (1) you make your payments by the due date each month and (2) this is the only balance on your account during the promo period.  If you have other balances on your account, this monthly payment will be added to the minimum payment applicable to those balances.  Regular account terms apply to non-promotional purchases. For new accounts: Purchase APR is 29.99. Minimum Interest Charge is $2. Existing cardholders should see their credit card agreement for their applicable terms. Subject to credit approval.</p><p>Rooms To Go requires a down payment equal to sales tax and delivery. </p>',
      },
    },
    marketingMessage: {
      markdown: {
        childMarkdownRemark: {
          html: 'No Interest until August 2023* marketing message',
        },
      },
    },
    threshold: '.00',
    numberOfMonths: '0',
    siteFriendlyLabel: 'No Interest until June 2021*',
  },
  {
    financeCode: '90015-0621',
    effectiveDate: '2019-06-03 00:00:00',
    creditEndsDate: '2021-06-01 00:00:00',
    downPaymentRequired: false,
    termsAndConditions: {
      childMarkdownRemark: {
        html:
          'Finance Plan:<br><br>Promotion Type: <strong>>No Interest until June 2021*</strong><br>Promotion Duration: <strong>Until  June 2021*</strong><br>Promotional APR: 0%<br>Purchase APR: 29.99% <br><br>Rooms To Go requires No Down Payment.<br><br><strong>No Interest/Equal Payments</strong><br>On promotional purchase, monthly payments equal to the promotional purchase divided equally by the number of months in the promotional period are required until expiration or termination of promotion, but no Finance Charges will be assessed during Promotion Duration indicated if all minimum monthly payments on account paid when due.<br>The Purchase APR as indicated applies to expired promotions and optional charges. If minimum monthly payments are 60 days past due, the promotion June be terminated and a Penalty APR June apply. Promotional purchases on existing accounts June not receive full benefit of promotion terms, including reduced APR if applicable, if account is subject to penalty pricing. Payments over the minimum will be applied as required by applicable law. If you receive a change in terms of penalty rate notice that becomes effective before the delivery/installation date of your purchase, the new Purchase APR or Penalty APR (as applicable) will apply to the purchase balance added to your account upon delivery/installation. Subject to approval by Synchrony Bank.<br>',
      },
    },
    priority: '2',
    promoMessage: {
      childMarkdownRemark: {
        html:
          '<p><b>24 MONTHS Interest Free Financing*</b></p><p>On purchases made with your Rooms To Go credit card through 6/17/19. Equal Monthly Payments Required until June 2021.*</p><p>Rooms To Go requires No Down Payment.</p><p>*Offer applies only to single-receipt qualifying purchases made on your Rooms To Go credit card made through 6/17/19. No interest will be charged on promo purchase and equal monthly payments are required equal to initial promo purchase amount divided equally by the number of months in promo period until promo is paid in full. The equal monthly payment will be rounded to the next highest whole dollar and may be higher than the minimum payment that would be required if the purchase was a non-promotional purchase. Regular account terms apply to non-promotional purchases. For new accounts: Purchase APR is 29.99%; Minimum Interest Charge is $2. Existing cardholders should see their credit card agreement for their applicable terms. Subject to credit approval.</p><p>Rooms To Go requires No Down Payment.</p>',
      },
    },
    threshold: '.00',
    numberOfMonths: '0',
    siteFriendlyLabel: '24 Months Interest Free Financing*',
  },
]
