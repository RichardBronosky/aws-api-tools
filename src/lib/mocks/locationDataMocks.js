export const mockLocation1 = {
  lat: 27.9836,
  long: -82.2756,
  city: 'Seffner',
  state: 'Florida',
  county: 'Hillsborough',
  zip: '33584',
  country_code: 'US',
  region: 'FL',
  delivery_zone: 'A',
  price_zone: '0',
  distribution_index: '1',
}

export const mockLocation2 = {
  lat: 33.92904,
  long: -84.3467,
  city: 'Dunwoody',
  state: 'Georgia',
  county: 'Fulton',
  zip: '30346',
  country_code: 'US',
  region: 'SE',
  delivery_zone: 'A',
  price_zone: '2',
  distribution_index: '10',
}
