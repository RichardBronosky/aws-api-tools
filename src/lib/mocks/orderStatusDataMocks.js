export const orderStatus = {
  AbleToReschedule: false,
  AdditionalDirections: '**** ALL DAY DELIVERY**** ****NO TIME FRAMES****',
  AffiliateData: '',
  Balance: '0.00',
  BillingAddress: {
    Address1: '1112 MARSH TRAIL CIR',
    Address2: '',
    City: 'ATLANTA',
    FirstName: 'SARAH',
    LastName: 'SMITH',
    State: 'GA',
    Zip: '30328',
  },
  Customer: {
    AllowText2AltPhoneNumber: false,
    AllowText2PhoneNumber: false,
    AltPhoneNumber: '',
    CMKey: '01*202/292-8491',
    CustomerID: 'SE006284958',
    Division: 'SE',
    FirstName: 'SARAH',
    LastName: 'SMITH',
    PhoneNumber: '202/292-8491',
  },
  DeliveryAddress: {
    Address1: '1112 MARSH TRAIL CIR',
    Address2: '',
    City: 'ATLANTA',
    Plus4: '',
    State: 'GA',
    Zip: '30328',
  },
  DeliveryCharge: '119.99',
  DeliveryDate: '2019-08-06T00:00:00.000Z',
  DeliveryType: 'E',
  Deposits: {
    Deposit: [
      {
        AuthorizationCode: '004547',
        AuthorizedAmount: '37.00',
        AuthorizedDate: '2019-08-04T00:00:00.000Z',
        CashCode: 'VA',
      },
    ],
  },
  Division: 'SE',
  EmailAddress: 'jamiamorrow@gmail.com',
  EstimatedDeliveryTimeFrame: '',
  FinDeposits: {
    Deposit: [
      {
        AuthorizationCode: '004547',
        AuthorizedAmount: '600.00',
        AuthorizedDate: '2019-08-04T00:00:00.000Z',
        CashCode: 'RGE',
      },
    ],
  },
  FinanceAccountNumber: '6019191263030424',
  FinancePlan: '90015-0121',
  FraudFlag: '',
  GCDeposits: {
    Deposit: [
      {
        AuthorizationCode: '004547',
        AuthorizedAmount: '1.95',
        AuthorizedDate: '2019-08-04T00:00:00.000Z',
        CashCode: 'GC',
      },
    ],
  },
  I95ind: false,
  LineItems: {
    LineItem: [
      {
        ChildItems: {
          LineItem: [
            {
              Quantity: 1,
              SelectedMonogramFee: 0,
              Sku: '36111639',
              UnitPrice: '193.24',
              UnitProtectionPrice: '0',
              UnitProtectionType: 'N',
            },
            {
              Quantity: 1,
              SelectedMonogramFee: 0,
              Sku: '36211631',
              UnitPrice: '250.76',
              UnitProtectionPrice: '0',
              UnitProtectionType: 'N',
            },
          ],
        },
        PackagePrice: '444',
        Quantity: 1,
        SelectedMonogramFee: 0,
        Sku: '3611163P',
        UnitPrice: '0',
        UnitProtectionPrice: '0',
        UnitProtectionType: 'N',
        DisplayName: 'Louie Gray Twin Loft Bed with Desk and Bookcase',
      },
      {
        ChildItems: null,
        Quantity: 1,
        SelectedMonogramFee: 0,
        Sku: '38096613',
        UnitPrice: '29',
        UnitProtectionPrice: '0',
        UnitProtectionType: 'N',
        DisplayName: 'Mattie Gray Chevron Bean Bag Chair',
      },
    ],
  },
  OrderDate: '2019-08-04T23:17:03.000Z',
  OrderMessage: 'OK',
  OrderNumber: '14000136I',
  OrderTotal: '638.95',
  PickUpData: {
    Phone: '',
    PickupAddress: { Address1: '', Address2: '', City: '', State: '', Zip: '' },
    PickupHours: { string: null },
    StoreName: '',
    StoreNumber: '980',
  },
  RemoteDeliveryDate: '0001-01-01T00:00:00.000Z',
  RemoteEstimatedDeliveryTimeFrame: '',
  ShipDate: '',
  Status: 'Open',
  StoreNumber: '980',
  TaxAmount: '45.96',
  TaxAreaId: '111211811',
  Active: true,
  DispatchTrack: 'https://fgo-rtg.dispatchtrack.com/track_order/f0d44a2/SE14000136I',
}

export const orderStatusList = [orderStatus]
