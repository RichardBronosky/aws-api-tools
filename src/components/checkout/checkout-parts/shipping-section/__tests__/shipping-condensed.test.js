import React from 'react'
import renderer from 'react-test-renderer'
import ShippingCondensed from '../shipping-condensed'
import { order1 } from '../../../../../lib/mocks/checkoutDataMocks'

describe('ShippingCondensed', () => {
  let testProps = {
    order: order1,
  }

  it('renders correctly when address looked up from QAS', () => {
    testProps.order.shippingAddress = {
      addressLookup: '1111 Test Street, Test City FL 33584',
      addressLookupSuccess: true,
      showAddressLookup: true,
    }

    const tree = renderer.create(<ShippingCondensed { ...testProps } />).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('renders correctly when address entered manually', () => {
    testProps.order.shippingAddress = {
      address1: '1111 Test Street',
      city: 'Test City',
      state: 'FL',
      zip: '33584',
      addressLookup: '',
    }

    const tree = renderer.create(<ShippingCondensed { ...testProps } />).toJSON()

    expect(tree).toMatchSnapshot()
  })
})
