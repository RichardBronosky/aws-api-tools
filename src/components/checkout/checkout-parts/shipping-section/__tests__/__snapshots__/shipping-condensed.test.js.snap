// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShippingCondensed renders correctly when address entered manually 1`] = `
<div
  className="shipping-condensed"
>
  <div
    className="left-info"
  >
    Tester Test
    <br />
    
    1111 Test Street
     undefined
    <br />
    Test City 
    FL 
    33584
  </div>
  <div
    className="right-info"
  >
    test@test.com
    <br />
    4040404040
    <br />
    
  </div>
</div>
`;

exports[`ShippingCondensed renders correctly when address looked up from QAS 1`] = `
<div
  className="shipping-condensed"
>
  <div
    className="left-info"
  >
    Tester Test
    <br />
    1111 Test Street
    <br />
     Test City FL 33584
  </div>
  <div
    className="right-info"
  >
    test@test.com
    <br />
    4040404040
    <br />
    
  </div>
</div>
`;
